### ISSUE TITLE
a11y: ensure search results state changes are accessible

### ISSUE THREAD

**Issue by schalkneethling** [schalkneethling/developer-toolchest#423]
**Describe the bug**

I am pretty sure that the way results are updated is not accessible or suboptimal for people using screen readers. Test the current experience and address any identified issues. Also, ensure that results are accessible to users using a keyboard. This will most likely involve moving focus appropriately and perhaps the use of `aria-live`.
